File: app/src/main/java/threadpoolexec/com/threadpoolexecutor/DI/NetworkComponent.java
package threadpoolexec.com.threadpoolexecutor.DI;

import javax.inject.Singleton;

import dagger.Component;
import threadpoolexec.com.threadpoolexecutor.services.DownloaderService;

@Singleton
@Component(modules = {NetworksModule.class})
public interface NetworkComponent {
    public void inject(DownloaderService service);
}

File: app/src/main/java/threadpoolexec/com/threadpoolexecutor/DI/NetworkComponent.kt
package threadpoolexec.com.threadpoolexecutor.DI

import dagger.Component
import threadpoolexec.com.threadpoolexecutor.services.DownloaderService
import javax.inject.Singleton

@Singleton
@Component(modules = [NetworksModule::class])
interface NetworkComponent {
    fun inject(service: DownloaderService)
}

